File: problem/power1/9-12/main.lua
-- main.lua: main file for "power" solution
--
-- Copyright (C) 2020-2021  Rudy Matela
-- All rights reserved.
--
--
-- This file is part of the CScx Online Judge.
--
-- This is appended to the submitted Lua program and tests the triple function.
--
-- The submitted file processes the standard input and this processes the
-- "in.txt" file.
io.input("in.txt")
x, y = io.read("*n", "*n")
while x and y do
  print(power(x,y))
  x, y = io.read("*n", "*n")
end

File: problem/power1/9-12/main.rb

# main.rb: main file for solutions to the power problem
#
# This is appended to the submitted Ruby program and tests the power function.
#
# The submitted file processes the standard input and this processes the
# "in.txt" file.
#
#
# Copyright (C) 2020-2021  Rudy Matela
# All rights reserved.
#
#
# This file is part of the CSCX Online Judge.
#
# This is to be appended to the submitted file.

# standard input is already processed
# as the submitted programmed is run when imported
File.foreach("in.txt") do |line|
  b, e = line.split.map { |x| x.to_i }
  puts power(b,e)
end
